### ISSUE TITLE
FUNSD dataset - empty key_dict

### ISSUE THREAD

**Issue by rrutmann** [ZeningLin/ViBERTgrid-PyTorch#17]
Hi,

first of all thanks for your great effort here!

I currently struggle to start a training on the FUNSD dataset. During the evaluation, I get the following error message:
Traceback (most recent call last):
  File "/home/rrutmann/pycharm_projects/ViGPTgrid/train_FUNSD.py", line 422, in <module>
    train(args)
  File "/home/rrutmann/pycharm_projects/ViGPTgrid/train_FUNSD.py", line 306, in train
    F1 = validate(
  File "/home/rrutmann/miniconda3/envs/vigptgrid/lib/python3.9/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
Traceback (most recent call last):
      File "/home/rrutmann/pycharm_projects/ViGPTgrid/train_FUNSD.py", line 422, in <module>
return func(*args, **kwargs)
  File "/home/rrutmann/pycharm_projects/ViGPTgrid/pipeline/train_val_utils.py", line 506, in validate
        curr_gt_str = key_dict[0][curr_class_name]train(args)

  File "/home/rrutmann/pycharm_projects/ViGPTgrid/train_FUNSD.py", line 306, in train
TypeError: 'NoneType' object is not subscriptable
    F1 = validate(
  File "/home/rrutmann/miniconda3/envs/vigptgrid/lib/python3.9/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/rrutmann/pycharm_projects/ViGPTgrid/pipeline/train_val_utils.py", line 506, in validate
    curr_gt_str = key_dict[0][curr_class_name]
TypeError: 'NoneType' object is not subscriptable

Indeed, key_dict is None. In the EPHOIE_DATASET and FUNSD_DATASET classes the variable full_key_dict is defined, which I assume is then later used for key_dict. But this part is missing in the FUNSD_DATASET class.

Could you please tell me how to handle this error?

**Comment by ZeningLin** [comment]
Hi,

I guess that the argument "eval_mode" was set to "seq_and_str" or "strcmp" in your configuration file. Just set it to "seqeval" and see whether it works or not.

The repository was initially designed for SROIE & EPHOIE dataset, which use a string comparison pipeline to compute the metrics. A key_dict that contains the gt key strings is included in the datasets' label file and is later used for metric computation. The eval_mode "strcmp" and "seq_and_str" are designed for this case. 

Later I tried to make the model work on the FUNSD dataset. The key_dict file, however, is not a part of the FUNSD labels. In this occasion, the only way to do evaluation is to use the seqeval library, corresponds to the "seqeval" mode. To keep the consistency of the arguments, I set the key_dict in FUNSD dataloader as None. That's why the error occured when using the "strcmp" or "seq_and_str" mode.
